src/main/resources/com/google/api/codegen/metadatagen/java/grpc/grpc_package.snip

```diff
@@ -51,12 +60,25 @@
       grpc {
         artifact = 'io.grpc:protoc-gen-grpc-java:{@metadata.grpcVersionBound.lower}'
       }
+      @switch metadata.packageType
+        @case "GRPC_CLIENT"
+          gapic {
+            path = locateGapicPluginMethod()
+          }
+      @end
     }
     generateProtoTasks {
       all()*.plugins {
         grpc {
           outputSubDir = 'java'
         }
+        @switch metadata.packageType
+        @case "GRPC_CLIENT"
+          gapic {
+            outputSubDir = 'java'
+            option '{@metadata.shortName}_gapic.yaml'
```

Comment:
I prefer to compose the full filename and put it in the model (I feel it is general enough).